doc/source/train/examples/pytorch/distributing-pytorch/README.md

```diff
@@ -0,0 +1,442 @@
+
+You can also click the "Metrics" tab on the navigation bar to view useful stats about your cluster, such as gpu utilization and metrics about ray [actors](https://docs.ray.io/en/latest/ray-core/actors.html) and [tasks](https://docs.ray.io/en/latest/ray-core/tasks.html).
+
+![Metrics Dashboard](images/metrics_dashboard.png)
+
+## Step 3: Speed up its data loading with Ray Data
+
+Let’s modify this example to load data with Ray Data instead of the native Torch DataLoader. With just a few modifications, you can offload data preprocessing to an independently scaling distributed Ray Data pipeline. See [here](https://docs.ray.io/en/latest/data/comparisons.html#how-does-ray-data-compare-to-other-solutions-for-ml-training-ingest) for a comparison between Ray Data and Torch data loading.
+
+First, create a [Ray Data Dataset](https://docs.ray.io/en/latest/data/key-concepts.html#datasets-and-blocks). Note the following about the code below:
+* You are now loading the data from the [Ansycale default storage bucket](https://docs.anyscale.com/configuration/storage/#anyscale-default-storage-bucket) instead of from PyTorch directly. 
+* Each row of a Ray Data Dataset is a dict so you need to access its contents by key.
+* Ray Data loads and transforms data lazily, which means that only the training workers materialize the data.
+
+
+```python
+STORAGE_PATH_PREFIX = os.environ.get("ANYSCALE_ARTIFACT_STORAGE", "artifact_storage")
+STORAGE_PATH = f"{STORAGE_PATH_PREFIX}/ray_summit_24_train_demo"
+
+def transform_cifar(row: dict):
+    # Define the torchvision transform.
+    transform = transforms.Compose([ToTensor(), Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+    row["image"] = transform(row["image"])
+    return row
+
+train_dataset = ray.data.read_parquet(f'{STORAGE_PATH}/cifar10-parquet/train').map(transform_cifar)
+test_dataset = ray.data.read_parquet(f'{STORAGE_PATH}/cifar10-parquet/test').map(transform_cifar)
+```
+
+Next, modify the training function you wrote earlier. Every difference from the previous script is highlighted and explained with a numbered comment; for example, “[1].”
+
+
+```python
+def train_func_per_worker(config: Dict):
+    lr = config["lr"]
+    epochs = config["epochs"]
+    batch_size = config["batch_size_per_worker"]
+
+    # [1] Prepare Dataloader for distributed training
+    # Shard the datasets among workers and move batches to the correct device
+    # =======================================================================
```

Comment:
Maybe elaborate more that the methods called are doing these? Otherwise it feels too auto-magical.

Reply:
Changed all of the numbered comments to mention what each function - ptal, thanks!